File: tools/lib/render.py
from __future__ import annotations

from typing import Any


def render_markdown(plan: dict[str, Any]) -> str:
    meta = plan.get("meta") or {}
    lines: list[str] = [
        f"# Test Plan - {meta.get('ticket_key', '')}",
        "",
        f"**Title:** {meta.get('title', '')}",
        f"**Prepared By:** {meta.get('prepared_by', '')}",
        f"**Date:** {meta.get('date', '')}",
        f"**Version:** {meta.get('version', '')}",
        "",
    ]
    for sec in plan.get("sections") or []:
        lines.append(f"## {sec.get('title')}")
        lines.append("")
        lines.append(sec.get("body_markdown") or "")
        lines.append("")
    return "\n".join(lines).strip() + "\n"

File: web/lib/render.ts
export function renderMarkdown(plan: Record<string, unknown>): string {
  const meta = (plan.meta as Record<string, string>) ?? {};
  const lines: string[] = [
    `# Test Plan - ${meta.ticket_key ?? ""}`,
    "",
    `**Title:** ${meta.title ?? ""}`,
    `**Prepared By:** ${meta.prepared_by ?? ""}`,
    `**Date:** ${meta.date ?? ""}`,
    `**Version:** ${meta.version ?? ""}`,
    "",
  ];
  const sections = (plan.sections as Array<{ title?: string; body_markdown?: string }>) ?? [];
  for (const sec of sections) {
    lines.push(`## ${sec.title ?? ""}`, "", sec.body_markdown ?? "", "");
  }
  return lines.join("\n").trim() + "\n";
}
